src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -306,5 +307,75 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
                 return ret;
             }
         }
+
+        /// <summary>
+        /// Writes a DataFrame into a CSV.
+        /// Follows pandas API.
+        /// </summary>
+        /// <param name="dataFrame"><see cref="DataFrame"/></param>
+        /// <param name="path">CSV file path</param>
+        /// <param name="separator">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="addIndexColumn">add one column with the row index</param>
+        /// <param name="encoding">The character encoding. Defaults to UTF8 if not specified</param>
+        public static void WriteCsv(DataFrame dataFrame, string path,
+                                   char separator = ',', bool header = true,
+                                   bool addIndexColumn = false, Encoding encoding = null)
+        {
+            using (FileStream csvStream = new FileStream(path, FileMode.Create))
+            {
+                WriteCsv(dataFrame: dataFrame, csvStream: csvStream,
+                           separator: separator, header: header,
+                           addIndexColumn: addIndexColumn, encoding: encoding);
+            }
+        }
+
+        /// <summary>
+        /// Writes a DataFrame into a CSV.
+        /// Follows pandas API.
+        /// </summary>
+        /// <param name="dataFrame"><see cref="DataFrame"/></param>
+        /// <param name="csvStream">stream of CSV data to be write out</param>
+        /// <param name="separator">column separator</param>
+        /// <param name="header">has a header or not</param>
+        /// <param name="addIndexColumn">add one column with the row index</param>
+        /// <param name="encoding">The character encoding. Defaults to UTF8 if not specified</param>
+        public static void WriteCsv(DataFrame dataFrame, Stream csvStream,
+                           char separator = ',', bool header = true,
+                           bool addIndexColumn = false, Encoding encoding = null)
+        {
+            using (StreamWriter csvFile = new StreamWriter(csvStream, encoding ?? Encoding.UTF8))
+            {
+                if (dataFrame != null)
+                {
+                    List<string> selectedColumnNames = dataFrame.Columns.GetColumnNames().ToList();
+
+                    if (addIndexColumn)
+                    {
+                        selectedColumnNames.Insert(0, "index");
+                    }
+
```

Comment:
I think a better approach is to use a `StringBuilder` object here and replace all calls to `string.Join` with `StringBuilder.Append`. If you then also added `CultureInfo`, you'll be able to pass it into the `StringBuilder` object and get the correct formatting for locales.

Reply:
I have replaced Join with StringBuilder for rows. (I didn't do it for the header)
I have removed the index column parameter.
I'm not totally convinced this is what you intended.